File: Easy/Java/13. Roman to Integer.java
class Solution {
    public int romanToInt(String s) {
        int num = 0;
        int ans = 0;

        for(int i = s.length() - 1; 0 <= i; i--){
            switch(s.charAt(i)){
                case 'I' : num = 1;
                    break;
                case 'V' : num = 5;
                    break;
                case 'X' : num = 10;
                    break;
                case 'L' : num = 50;
                    break;
                case 'C' : num = 100;
                    break;
                case 'D' : num = 500;
                    break;
                case 'M' : num = 1000;
                    break;
            }
            if(num * 4 < ans)
                ans -= num;
            else
                ans += num;
        }

        return ans;
    }
}

File: Easy/Python/13. Roman to Integer.py
class Solution(object):
    def romanToInt(self, s):
        num = 0
        ans = 0

        for i in range(len(s) - 1, -1, -1):
            if s[i] == 'I':
                num = 1
            elif s[i] == 'V':
                num = 5
            elif s[i] == 'X':
                num = 10
            elif s[i] == 'L':
                num = 50
            elif s[i] == 'C':
                num = 100
            elif s[i] == 'D':
                num = 500
            elif s[i] == 'M':
                num = 1000
            
            if num * 4 < ans:
                ans -= num
            else:
                ans += num

        return ans
